packages/easy-testing/src/DataProvider/StaticFixtureFinder.php

```diff
@@ -18,55 +17,61 @@ final class StaticFixtureFinder
 {
     public static function yieldDirectory(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $fileInfos = self::findFilesInDirectory($directory, $suffix);
-        foreach ($fileInfos as $fileInfo) {
+        $finder = self::findFilesInDirectory($directory, $suffix);
+        foreach ($finder as $fileInfo) {
             yield [new SmartFileInfo($fileInfo->getRealPath())];
         }
     }
 
     public static function yieldDirectoryExclusively(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $fileInfos = self::findFilesInDirectoryExclusively($directory, $suffix);
-        foreach ($fileInfos as $fileInfo) {
+        $finder = self::findFilesInDirectoryExclusively($directory, $suffix);
+        foreach ($finder as $fileInfo) {
             yield [new SmartFileInfo($fileInfo->getRealPath())];
         }
     }
 
-    /**
-     * @return SplFileInfo[]
-     */
-    private static function findFilesInDirectory(string $directory, string $suffix): array
+    public static function yieldDirectoryByFileName(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $finder = Finder::create()->in($directory)->files()->name($suffix);
-        $fileInfos = iterator_to_array($finder);
+        $finder = self::findFilesInDirectory($directory, $suffix);
+        foreach ($finder as $fileInfo) {
+            yield $fileInfo->getRelativePathname() => [new SmartFileInfo($fileInfo->getRealPath())];
+        }
+    }
+
+    public static function yieldDirectoryExclusivelyByFileName(string $directory, string $suffix = '*.php.inc'): Iterator
+    {
+        $finder = self::findFilesInDirectoryExclusively($directory, $suffix);
+        foreach ($finder as $fileInfo) {
+            yield $fileInfo->getRelativePathname() => [new SmartFileInfo($fileInfo->getRealPath())];
+        }
+    }
 
-        return array_values($fileInfos);
+    private static function findFilesInDirectory(string $directory, string $suffix): Finder
+    {
+        return Finder::create()
+            ->in($directory)
+            ->files()
+            ->name($suffix);
     }
 
-    /**
-     * @return SplFileInfo[]
-     */
-    private static function findFilesInDirectoryExclusively(string $directory, string $suffix): array
+    private static function findFilesInDirectoryExclusively(string $directory, string $suffix): Finder
     {
         self::ensureNoOtherFileName($directory, $suffix);
 
-        $finder = Finder::create()->in($directory)
+        return Finder::create()->in($directory)
             ->files()
             ->name($suffix);
-
-        $fileInfos = iterator_to_array($finder->getIterator());
-        return array_values($fileInfos);
```

Comment:
This contract should not be changed. Hiding files behind Finder iterator gives space for bugs.

Reply:
How does it give space for bugs? Not sure I really get the issue with it, but I'll change it :)

Reply:
E.g. the implicit key relative path. It's relative to different directory root

Reply:
Thanks :+1: